### PR TITLE
SCIM: Add feature flag

### PR DESCRIPTION
**Pull request by linoman**
**What is this feature?**

Adds feature flag for SCIM enablement.

**Why do we need this feature?**

To enable SCIM integration.

**Who is this feature for?**

IAM

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/identity-access-team/issues/1001

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/contribute/release-notes/#how-to-determine-if-content-belongs-in-whats-new), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/contribute/release-notes/) doc.

### CODE REVIEW

File: pkg/services/featuremgmt/registry.go [hunk 1]
```diff
@@ -1602,6 +1602,12 @@ var (
 			Stage:       FeatureStageExperimental,
 			Owner:       grafanaOSSBigTent,
 		},
+		{
+			Name:        "enableScim",
```
Comment (kalleep) [inline pkg/services/featuremgmt/registry.go:1606]:
```suggestion
			Name:        "enableSCIM",
```

Comment (kalleep) [reply pkg/services/featuremgmt/registry.go:1606]: We should use uppercase letters for acronyms

File: pkg/services/featuremgmt/toggles_gen.json [hunk 2]
```diff
@@ -1231,6 +1231,31 @@
...
+        "resourceVersion": "1730976932709",
+        "creationTimestamp": "2024-11-07T10:55:32Z",
+        "deletionTimestamp": "2024-11-07T11:54:44Z"
+      },
+      "spec": {
+        "description": "Enables SCIM support for user and group management",
+        "stage": "experimental",
+        "codeowner": "@grafana/identity-access-team"
+      }
+    },
```
Comment (gamab) [inline pkg/services/featuremgmt/toggles_gen.json:1258]: Left over from the initial name?
```suggestion
```

**Review by gamab — APPROVED** [review]
I think you can remove the deletion entry in `toggles_gen.json`

File: pkg/services/featuremgmt/toggles_gen.json (hunk 2 above)
Comment (linoman) [reply pkg/services/featuremgmt/toggles_gen.json:1258]: Ohh wow I missed this, but you are right! Thanks for the catch

**Review by kalleep — APPROVED** [review]

File: pkg/services/featuremgmt/toggles_gen.json (hunk 2 above)
Comment (linoman) [reply pkg/services/featuremgmt/toggles_gen.json]: Removed.

**Pull request merged**